src/topology/sequences.lean

```diff
@@ -168,6 +169,240 @@ instance : sequential_space α :=
     apply hU.tendsto hu.2,
   end⟩
 
+
 end first_countable_topology
 
 end topological_space
+
+section seq_compact
+open topological_space topological_space.first_countable_topology
+variables [topological_space α]
+
+/-- A set `s` is sequentially compact if every sequence taking values in `s` has a
+converging subsequence. -/
+def seq_compact (s : set α) :=
+  ∀ ⦃u : ℕ → α⦄, (∀ n, u n ∈ s) → ∃ (x ∈ s) (φ : ℕ → ℕ), strict_mono φ ∧ tendsto (u ∘ φ) at_top (𝓝 x)
```

Comment:
I think it would be better to ask for `tendsto φ at_top at_top` in the definition. This way it will be easier to verify this definition for a particular set `s`. Then we can have a lemma `seq_compact.exists_strict_mono` that provides a `strict_mono` function.

BTW, do you actually need `strict_mono` in some proofs?

Reply:
I understand what you wrote but I'm not sure it would be worth the refactoring trouble. I don't think we'll have that many proofs of sequential compactness. There should be many more *uses* of sequential compactness, where you want the strong conclusion. And we already have the required lemmas to go from the weak form to the strong one if you really need it.